### PR TITLE
For #72: Introduce error mechanics.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #72: Introduce error mechanics:
- Added `Error` and `Errors` interfaces
- Changed `User` to return user errors
- Created tests for `DyError` and `DyErrors` implementations
- Left puzzles for continue implementation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#75](https://github.com/yegor256/wring/issues/75) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#75](https://github.com/yegor256/wring/issues/75) is assigned to @pinaf[/z](https://www.0crat.com/u/pinaf), here is [why](https://www.0crat.com/footprint/C7FCB1EQN/a3e5dc0e-0466-43da-af76-ee5833989ca3); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/io/wring/dynamo/DyErrors.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import io.wring.model.Errors;
+import java.io.IOException;
+
+/**
+ * Errors stored in Dynamo database.
+ *
+ * @author Paulo Lobo (pauloeduardolobo@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ * @todo #72:30min Implement DyError and DyErrors methods. These
```
Comment (pinaf) [inline src/main/java/io/wring/dynamo/DyErrors.java]: @paulodamaso I believe you mean `classes` here instead of `methods`.

File: src/main/java/io/wring/model/Error.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     * @return Xembly directives
+     * @throws IOException If fails
+     */
+    Iterable<Directive> asXembly() throws IOException;
+
+    /**
+     * Delete it.
+     * @throws IOException If fails
+     */
+    void delete() throws IOException;
```
Comment (pinaf) [inline src/main/java/io/wring/model/Error.java]: @paulodamaso why do we want to delete errors?

File: src/main/java/io/wring/agents/Exec.java [hunk 3]
```diff
@@ -54,6 +54,9 @@
  * @author Yegor Bugayenko (yegor256@gmail.com)
  * @version $Id$
  * @since 0.13
+ * @todo #72:30min Errors occurred during execution must be saved and later
+ *  showed to user. Wire Error and Errors implementations in Exec flow so these
+ *  errors are properly logged once they happen. Cover this with tests.
```
Comment (pinaf) [inline src/main/java/io/wring/agents/Exec.java]: @paulodamaso how about a puzzle for listing the errors and displaying them somewhere?

File: src/main/java/io/wring/dynamo/DyErrors.java (hunk 1 above)
Comment (paulodamaso) [reply src/main/java/io/wring/dynamo/DyErrors.java]: @pinaf Actually I was meaning "methods from DyError and DyErrors", but I'll stick to your suggestion since it describes the issue better.

File: src/main/java/io/wring/model/Error.java (hunk 2 above)
Comment (paulodamaso) [reply src/main/java/io/wring/model/Error.java]: @pinaf After we see the error on error page it  would be deleted, just like en `Event`

File: src/main/java/io/wring/agents/Exec.java (hunk 3 above)
Comment (paulodamaso) [reply src/main/java/io/wring/agents/Exec.java]: @pinaf That's exactly what this puzzle is about. Let me reword it to become more clear.

**Comment by paulodamaso** [comment]
@pinaf Made some changes, please take a look now.

**Comment by pinaf** [comment]
@paulodamaso thanks

**Comment by pinaf** [comment]
@carlosmiranda good to merge

**Comment by carlosmiranda** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @pinaf[/z](https://www.0crat.com/u/pinaf), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#75](https://github.com/yegor256/wring/issues/75) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by ypshenychka** [comment]
@pinaf According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by pinaf** [comment]
@ypshenychka I confirm.

**Comment by ypshenychka** [comment]
@pinaf thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/yegor256/wring/issues/75#issuecomment-434224231))

@ypshenychka There is no quality review for [#75](https://github.com/yegor256/wring/issues/75), no performer

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @pinaf[/z](https://www.0crat.com/u/pinaf)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
